doc/source/whatsnew/v1.2.5.rst

```diff
@@ -34,8 +34,7 @@ Bug fixes
 
 Other
 ~~~~~
-
--
+- Loosen XLS signatures used in :func:`read_excel` to determine if the `xlrd` engine should be used (:issue:`41225`)
```

Comment:
Please move to 1.3, I think we have a section for IO excel or something like that